include/fiction/algorithms/simulation/sidb/random_layout_generator.hpp

```diff
@@ -0,0 +1,200 @@
+//
+// Created by Jan Drewniok on 05.04.23.
+//
+
+#ifndef FICTION_RANDOM_LAYOUT_GENERATOR_HPP
+#define FICTION_RANDOM_LAYOUT_GENERATOR_HPP
+
+#include "fiction/algorithms/path_finding/distance.hpp"
+#include "fiction/io/write_sqd_layout.hpp"
+#include "fiction/layouts/cell_level_layout.hpp"
+#include "fiction/technology/cell_technologies.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/technology/sidb_nm_position.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+
+#include <iostream>
+#include <vector>
+
+namespace fiction
+{
+/**
+ * This struct stores the parameters for the *generate_random_layouts* algorithm.
+ */
+template <typename Lyt>
+struct random_layout_params
+{
+    /**
+     * Aspect ratio of the layout (i.e. the size of the layout).
+     */
+    typename Lyt::aspect_ratio max_coordinate;
+    /**
+     * Number of SiDBs that are placed on the layout.
+     */
+    uint64_t number_placed_sidbs = 0;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than the minimal_spacing.
+     */
+    bool prevent_positive_charges = true;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than this value (2 cells as
+     * Euclidean distance by default).
```

Comment:
I'm not sure that I understand this. How far are 2 cells by Euclidean distance?

Reply:
I mean it depends on the cells, right? cell1{0,0} and cell{2,0} have a distance of 2. Or what do you mean?

Reply:
I meant that the Euclidean distance does not provide a distance in cells, but a unit-less distance value.